src/coreclr/inc/stresslog.h

```diff
@@ -544,25 +544,45 @@ inline BOOL StressLog::LogOn(unsigned facility, unsigned level)
 #pragma warning(disable:4200 4201)					// don't warn about 0 sized array below or unnamed structures
 #endif
 
-// The order of fields is important.  Keep the prefix length as the first field.
-// And make sure the timeStamp field is naturally aligned, so we don't waste
-// space on 32-bit platforms
-struct StressMsg {
-    static const size_t formatOffsetBits = 26;
-    union {
-        struct {
-            uint32_t numberOfArgs  : 3;                   // at most 7 arguments here
-            uint32_t formatOffset  : formatOffsetBits;    // offset of string in mscorwks
-            uint32_t numberOfArgsX : 3;                   // extend number of args in a backward compat way
-        };
-        uint32_t fmtOffsCArgs;    // for optimized access
-    };
-    uint32_t facility;                      // facility used to log the entry
-    uint64_t timeStamp;                     // time when mssg was logged
-    void*     args[0];                      // size given by numberOfArgs
+// The order of fields is important.  Ensure that we minimize padding
+// to fit more messages in a chunk.
+struct StressMsg
+{
+    static const size_t formatOffsetLowBits = 26;
+    static const size_t formatOffsetHighBits = 13;
+
+    uint32_t numberOfArgsLow  : 3;                   // at most 7 arguments here
```

Comment:
This is getting pretty complicated. Do we really need to maintain the best-effort backward compatibility here?

Reply:
We can shift around some of the structure, rev the version, and keep most of the complexity in SOS only for its back-compat.

Reply:
I think it would be better - if the SOS side of the fix is not too complicated.

Reply:
Updated the format to be cleaner and the corresponding change in the diagnostics repo to support reading both this format and the prior one.